moviebot/nlu/text_processing.py

```diff
@@ -5,34 +5,39 @@
 """
 
 import string
-from typing import List, Optional, Text
+from typing import List, Optional
 
 from nltk.corpus import stopwords
 from nltk.stem import WordNetLemmatizer
 
 
 class Span:
-    """Subpart of an utterance.
-
-    Contains mapping of start and end positions in the original
-    utterance.
-    """
-
     def __init__(
         self,
-        text: Text,
+        text: str,
         start: int,
         end: Optional[int] = None,
-        lemma: Optional[Text] = None,
+        lemma: Optional[str] = None,
     ) -> None:
+        """Subpart of an utterance.
+
+        Contains mapping of start and end positions in the original
+        utterance.
+
+        Args:
+            text: Raw token string.
+            start: Start of the token in utterance.
+            end: End of token in utterance. Defaults to None.
+            lemma: Lemmatized token string. Defaults to None.
+        """
         self.text = text
 
         self.start = start
         self.end = end if end else start + len(text)
 
         self.lemma = lemma if lemma else text
 
-    def overlaps(self, other) -> bool:
+    def overlaps(self, other: "Span") -> bool:
```

Comment:
Is there a reason to put a type in "" here?

Reply:
Because Span does not exist yet. It is a reference to the class where the typed method is defined.